nomenclature/__init__.py

```diff
@@ -52,3 +54,44 @@ def create_yaml_from_xlsx(source, target, sheet_name, col, attrs=None):
     SPECIAL_CODELIST.get(col.lower(), CodeList).read_excel(
         name="", source=source, sheet_name=sheet_name, col=col, attrs=attrs
     ).to_yaml(target)
+
+
+def parse_model_registration(source, file_name):
+    """Parses a model registration file and writes the definitions & mapping yaml files
+
+    Parameters
+    ----------
+    source : str, path, file-like object
+        Path to xlsx model registration file.
+    file_name : str
+        Model-identifier part of the yaml file names.
+    """
+    model = pd.read_excel(source, sheet_name="Model", usecols="B", nrows=1).iloc[0, 0]
+    regions = pd.read_excel(source, sheet_name="Regions", header=2)
+
+    native = "Native region (as submitted)"
+    rename = "Native region (as shown in the Explorer)"
+    sep = "\n  - "
+
+    common_region_groups = [r for r in regions.columns if r not in [native, rename]]
+
+    # write region definitions file
+    with open(f"definitions/region/model_native_regions/{file_name}.yaml", "w") as file:
+        file.write(f"- {model}:{sep}" + sep.join(regions[rename].values))
+
+    # write mappings file
+    with open(f"mappings/{file_name}.yaml", "w") as file:
+        file.write(f"model:{sep}{model}\n")
```

Comment:
Similar idea as above, I would create a dictionary that represents the model mapping and then dump directly to yaml.
Alternatively, we could also use the `RegionAggregationMapping` class directly, although that might be a bit overkill.
```python
model_mapping = {"model": model}
model_mapping["native_regions"]=[{} for _, row in regions.iterrows()]
model_mapping["common_regions"]=...
```